tests/misc/CachingTest.py

```diff
@@ -0,0 +1,86 @@
+import os
+import unittest
+
+from pyprint.NullPrinter import NullPrinter
+
+from coalib.misc.Caching import (
+    get_config_data_path, pickle_load, pickle_dump, add_to_changed_files,
+    add_new_files_since_last_run, get_last_coala_run_time,
+    update_last_coala_run_time, verify_file_integrity)
+from coalib.output.printers.LogPrinter import LogPrinter
+
+
+class CachingTest(unittest.TestCase):
+
+    def test_pickling(self):
+        test_data = {"answer": 42}
+
+        pickle_dump("test_file", test_data)
+        self.assertEqual(pickle_load("test_file"), test_data)
+        os.remove(get_config_data_path("test_file"))
+
+        self.assertEqual(pickle_load("nonexistant_file"), None)
+        self.assertEqual(pickle_load("nonexistant_file", fallback=42), 42)
```

Comment:
I've changed things a bit, using the `get_tags_dir` from `coalib.output.Tagging` to get the file path. The implication of this is that all functions now need `log_printer`. It seems too forced (and bloated, defeating the whole purpose of doctests being clean) to import a NullPrinter for each doctest. Still go ahead with doctests?